component/src/main/java/org/ballerinalang/compiler/FTPMonitorServiceCompilerPlugin.java

```diff
@@ -0,0 +1,135 @@
+                                + " parameter allow in the resource signature.");
+            }
+            final BType type = parameters.get(0).getTypeNode().type;
+            if (type.getKind().equals(TypeKind.STRUCT)) {
+                if (type instanceof BStructType) {
+                    BStructType event = (BStructType) type;
+                    if (!"ftp".equals(event.tsymbol.pkgID.name.value) || !FTP_SERVER_EVENT
+                            .equals(event.tsymbol.name.value)) {
+                        dlog.logDiagnostic(Diagnostic.Kind.ERROR, parameters.get(0).getPosition(),
+                                "Parameter should be of type - ftp:" + FTP_SERVER_EVENT);
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void process(EndpointNode endpointNode, List<AnnotationAttachmentNode> annotations) {
+        final ExpressionNode configurationExpression = endpointNode.getConfigurationExpression();
+        if (NodeKind.RECORD_LITERAL_EXPR.equals(configurationExpression.getKind())) {
+            BLangRecordLiteral recordLiteral = (BLangRecordLiteral) configurationExpression;
+            boolean isNonEmptyPath = false;
+            boolean isNonEmptyHost = false;
+            for (BLangRecordLiteral.BLangRecordKeyValue config : recordLiteral.getKeyValuePairs()) {
+                final String key = ((BLangSimpleVarRef) config.getKey()).variableName.value;
+                if (config.getValue() instanceof BLangLiteral) {
+                    final Object value = ((BLangLiteral) config.getValue()).getValue();
+                    switch (key) {
+                        case FtpConstants.ENDPOINT_CONFIG_PATH:
+                            if (value != null) {
+                                if (!value.toString().isEmpty()) {
+                                    isNonEmptyPath = true;
+                                }
+                            }
+                            break;
+                        case FtpConstants.ENDPOINT_CONFIG_HOST:
+                            if (value != null) {
+                                if (!value.toString().isEmpty()) {
+                                    isNonEmptyHost = true;
+                                }
+                            }
+                            break;
+                        default:
+                            // Do nothing.
+                    }
+                }
+            }
+            if (!isNonEmptyPath) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, endpointNode.getPosition(),
+                        "Cannot create FTP Listener without path.");
+            }
+            if (!isNonEmptyHost) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, endpointNode.getPosition(),
+                        "Cannot create FTP Listener without host.");
+                throw new BallerinaException("Cannot create FTP Listener without host.");
```

Comment:
if ((BLangLiteral) config.getValue()).getValue() returns null we should throw this exception right ?

Reply:
I have updated the code to throw error before the switch statement if the value is null.